extensions-core/stats/src/main/java/io/druid/query/aggregation/teststats/PvaluefromZscorePostAggregator.java

```diff
@@ -0,0 +1,167 @@
+import com.google.common.collect.Sets;
+import io.druid.query.Queries;
+import io.druid.query.aggregation.AggregatorFactory;
+import io.druid.query.aggregation.PostAggregator;
+import io.druid.query.aggregation.post.ArithmeticPostAggregator;
+import io.druid.query.aggregation.post.PostAggregatorIds;
+import io.druid.query.cache.CacheKeyBuilder;
+import org.apache.commons.math3.distribution.NormalDistribution;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Set;
+
+@JsonTypeName("pvalue2tailedZtest")
+public class PvaluefromZscorePostAggregator implements PostAggregator
+{
+  private final String name;
+  private final PostAggregator zScore;
+
+  @JsonCreator
+  public PvaluefromZscorePostAggregator(
+      @JsonProperty("name") String name,
+      @JsonProperty("zScore") PostAggregator zScore
+  )
+  {
+    Preconditions.checkNotNull(
+        name,
+        "Must have a valid, non-null post-aggregator"
+    );
+    Preconditions.checkNotNull(
+        zScore,
+        "Must have a valid, non-null post-aggregator"
+    );
+    this.name = name;
+    this.zScore = zScore;
+  }
+
+  @Override
+  public Set<String> getDependentFields()
+  {
+    Set<String> dependentFields = Sets.newHashSet();
+
+    dependentFields.addAll(zScore.getDependentFields());
+
+    return dependentFields;
+  }
+
+  @Override
+  public Comparator getComparator()
+  {
+    return ArithmeticPostAggregator.DEFAULT_COMPARATOR;
+  }
+
+  @Override
+  public Object compute(Map<String, Object> combinedAggregators)
+  {
+
+    double zScoreValue = ((Number) zScore.compute(combinedAggregators))
+        .doubleValue();
+
+    zScoreValue = Math.abs(zScoreValue);
+    return 2 * (1 - cumulativeProbability(zScoreValue));
+  }
+
+  private double cumulativeProbability(double x)
+  {
+    try {
+      NormalDistribution normDist = new NormalDistribution();
+      return normDist.cumulativeProbability(x);
+    }
+    catch (IllegalArgumentException ex) {
+      return Double.NaN;
+    }
+  }
+
+  @Override
+  @JsonProperty
+  public String getName()
+  {
+    return name;
+  }
+
+  @Override
+  public PostAggregator decorate(Map<String, AggregatorFactory> aggregators)
+  {
+    return new PvaluefromZscorePostAggregator(
+        name,
+        Iterables.getOnlyElement(Queries.decoratePostAggregators(
+            Collections.singletonList(zScore), aggregators))
+    );
+  }
+
+  @JsonProperty
+  public PostAggregator getZscore()
+  {
+    return zScore;
+  }
+
+  @Override
+  public boolean equals(Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    PvaluefromZscorePostAggregator that = (PvaluefromZscorePostAggregator) o;
+
+    if (!name.equals(that.name)) {
+      return false;
+    }
+
+    return (zScore.equals(that.zScore));
+  }
+
+  @Override
+  public int hashCode()
+  {
+    int result = name != null ? name.hashCode() : 0;
```

Comment:
Please remove unnecessary null check for name. It's already checked in the constructor.